jest/globals.js

```diff
@@ -0,0 +1,307 @@
+    window.hkdfH = sjcl.codec.base64.fromBits(sjcl.codec.bytes.toBits(hBytes));
+
+    window.LISTENER_URLS = workflow.__get__("LISTENER_URLS");
+
+    return workflow
+}
+
+window.localStorageMock = {
+    getItem: key => JSON.parse(localStorageItems[key]),
+    setItem: (key, value) => localStorageItems[key] = JSON.stringify(value),
+    clear: () => localStorageItems = {},
+    removeItem: (key) => localStorageItems[key] = undefined,
+};
+
+window.updateIconMock = jest.fn();
+window.updateBrowserTabMock = jest.fn();
+window.reloadBrowserTabMock = jest.fn();
+window.validateRespMock = jest.fn();
+
+window.CACHED_COMMITMENTS_STRING = "cached-commitments";
+window.clearCachedCommitmentsMock = () => setMock(CACHED_COMMITMENTS_STRING, null);
+
+window.setSpendFlagMock = (key, value) => setMock(key, value);
+window.getSpendFlagMock = (key) => getMock(key)
+
+window.getMock = (key) => JSON.parse(localStorage.getItem(key));
+window.setMock = (key, value) => localStorage.setItem(key, JSON.stringify(value));
+window.clearLocalStorage = () => localStorage.clear()
+
+window.bypassTokens = (config_id) => `bypass-tokens-${config_id}`;
+window.bypassTokensCount = (config_id) => `bypass-tokens-count-${config_id}`;
+
+/* mock XHR implementations */
+window.mockXHR = (_xhr) => {
+    _xhr.open = function (method, url) {
+        _xhr.method = method;
+        _xhr.url = url;
+    };
+    _xhr.requestHeaders = new Map();
+    _xhr.getRequestHeader = function (name) {
+        return _xhr.requestHeaders[name];
+    }
+    _xhr.setRequestHeader = function (name, value) {
+        _xhr.requestHeaders[name] = value;
+    }
+    _xhr.overrideMimeType = jest.fn();
+    _xhr.body;
+    _xhr.send = jest.fn();
+    _xhr.onreadystatechange = function () {
+    };
+}
+
+const storedTokens = `[{
+    "data": [24, 62, 56, 102, 76, 127, 201, 111, 161, 218, 249, 109, 34, 122, 160, 219, 93, 186, 246, 12, 178, 249, 241, 108, 69, 181, 77, 140, 158, 13, 216, 184],
+    "point": "BPzFsXoTj3RkTpcuyUUMV6eBvFkczB06iI7XNe5F65OluItpyKd0Eb9/9RFUAHiX5k7yMFVICUzCNyoRkOKUjU0=",
+    "blind": "0x46af9794d53f040607a35ad297f92aef6a9879686279a12a0a478b2e0bde9089"
+}, {
+    "data": [131, 120, 153, 53, 158, 58, 11, 155, 160, 109, 247, 176, 176, 153, 14, 161, 150, 120, 43, 180, 188, 37, 35, 75, 52, 219, 177, 16, 24, 101, 241, 159],
+    "point": "BLJ+ClrY1PkS+2hOd86eMFHYaJOFFPYmUwMEFVQJgaBgPl5bRvxN1ecSCmdX8Q7NKoWxRtbo12o2FLfN9a0HAro=",
+    "blind": "0xd475b86c84c94586503f035911388dd702f056472a755e964cbbb3b58c76bd53"
+}]`;
+const testG = "BOidEuO9HSJsMZYE/Pfc5D+0ELn0bqhjEef2O0u+KAw3fPMHHXtVlEBvYjE5I/ONf9SyTFSkH3mLNHkS06Du6hQ=";
+const testH = "BHOPNAWXRi4r/NEptOiLOp8MSwcX0vHrVDRXv16Jnowc1eXXo5xFFKIOI6mUp8k9/eca5VY07dBhAe8QfR/FSRY=";
+const testDevG = "BIpWWWWFtDRODAHEzZlvjKyDwQAdh72mYKMAsGrtwsG7XmMxsy89gfiOFbX3RZ9Ik6jEYWyJB0TmnWNVeeZBt5Y=";
+const testDevH = "BKjGppSCZCsL08YlF4MJcml6YkCglMvr56WlUOFjn9hOKXNa0iB9t8OHXW7lARIfYO0CZE/t1SlPA1mXdi/Rcjo=";
```

Comment:
These variables `testG`, `testH`, `testDevG`, `testDevH` appaear to be duplicated.